miner/worker.go

```diff
@@ -210,9 +218,18 @@ func (miner *Miner) generateWork(params *generateParams, witness bool) *newPaylo
 	if err != nil {
 		return &newPayloadResult{err: err}
 	}
+
+	var rates common.ExchangeRates
+	if shouldFetchRates {
+		rates, err = miner.backend.CeloAPIBackend().GetExchangeRates(context.Background(), rpc.BlockNumberOrHashWithHash(block.Header().ParentHash, false))
+		if err != nil {
+			return &newPayloadResult{err: err}
+		}
+	}
+
 	return &newPayloadResult{
 		block:    block,
-		fees:     totalFees(block, work.receipts),
+		fees:     totalFees(block, work.receipts, rates),
```

Comment:
The exchange rates are already present in the returned environment from `prepareWork`, so you can simply do:


```suggestion
		fees:     totalFees(block, work.receipts, work.feeCurrencyContext.ExchangeRates),

```
and **remove all other changes in this file.**

Reply:
We might have to double check to make sure this can't be `nil` or the cases were it is nil don't require the rates